Documentation/proposals/202309-controller-id.md

```diff
@@ -0,0 +1,65 @@
+# Controller-ID Proposal
+
+* Owners:
+  * [danielmellado](https://github.com/danielmellado)
+* Related Tickets:
+  * [#4281](https://github.com/prometheus-operator/prometheus-operator/issues/4281)
+  * [#4498](https://github.com/prometheus-operator/prometheus-operator/pull/4498)
+* Other Docs:
+  * n/a
+
+# Introduction
+
+This proposal aims to implement a solution to support multiple cluster-level
+Prometheus instances running concurrently without conflicting over the same
+custom resources.
+
+This issue can significantly impact use cases where multiple Prometheus operator
+instances run in parallel.
+
+# Why
+
+Currently, we encounter issues when different users deploy different instances
+of the Prometheus operator, that will try to reconcile the same resources.
+
+In the worst-case scenario, these operators may not only compete for ownership
+of the CRD resources but also attempt to rewrite or redeploy different versions
+of the CRD, causing disruptions to all pods.
+
+The remediation for this scenario, where users deploy their Prometheus operator
+instances in parallel, involves using one of the many CLI arguments, such as
+`--deny-namespaces`, `--namespaces`, `--prometheus-instance-selector`, or
+`prometheus-instance-namespaces`.
+
+# How
+
+After some research, we have identified a potential solution already
+implemented by the
+[zalando/postgres-operator](https://github.com/zalando/postgres-operator). When
+an operator is configured with a specific "controller ID" value, it will only
+reconcile resources that have a matching "controller ID" annotation.
+
+Conversely, if the operator is not configured with a "controller ID," it will
+skip all resources that have a "controller ID" annotation. More details can be
+found in the
+[zalando/postgres-operator documentation](https://github.com/zalando/postgres-operator/blob/master/docs/administrator.md#operators-with-defined-ownership-of-certain-postgres-clusters).
+
+# Goals
+
+* Guarantee that a custom resource will be managed by a specific Prometheus
+  operator instance.
+
+## Audience
+
+This proposal is relevant to the following audience:
+
+* Users who provide Prometheus as a service and want to run multiple Prometheus
+  operator instances in different namespaces.
+* Users seeking to mitigate the impact of rogue Prometheus instances.
+
+# Alternatives
+
+Although the initial discussion for this proposal considered adding an owner
+reference within the scope, the
+[ControllerRef](https://github.com/kubernetes/design-proposals-archive/blob/acc25e14ca83dfda4f66d8cb1f1b491f26e78ffe/api-machinery/controller-ref.md)
+model does not directly address this problem.
```

Comment:
```suggestion
model does not directly address this problem because the `ControllerRef` model solves the problem of controllers that fight over controlled objects due to overlapping selectors (e.g. a ReplicaSet fighting with a ReplicationController over Pods because both controllers have label selectors that match those Pods)
```